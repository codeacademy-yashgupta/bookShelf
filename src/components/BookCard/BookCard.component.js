/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import './BookCard.css';
import CardReaction from '../CardReaction/CardReaction.component';

class BookCard extends Component {
  render() {
    return (
      <div className="card-body">
        <div className="card-image">
          <img alt="" src="./images/cover_image.jpeg" />
        </div>
        <div className="card-text-title">
          Harry Potter and the Sorcerers Stone (Harry Potter, #1)
        </div>
        <div className="card-footer">
          <div className="card-text-rating">
          4.5
          </div>
          <div className="card-like">
            <CardReaction />
          </div>
        </div>
      </div>
    );
  }
}
export default BookCard;
