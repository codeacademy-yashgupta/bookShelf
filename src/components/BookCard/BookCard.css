.card-body {
  display: flex;
  flex-direction: column;
  width: 320px;
  height: 380px;
  margin: 20px;
  border: 1px solid black;
  min-width: 300px;
  background-color: white;
  color: orange;

}
img {
  width: 320px;
}
.card-text-title {
  padding: 10px;
  height: 80px;
  font-size: 22px;
  text-align: center;
  font-weight: bold;
}
.card-text-rating {
  height: 30px;
  padding: 10px;
}
.card-image {
  width: 320px;
  overflow: hidden;
  height: 200px;
}
.card-footer {
  display: flex;
  flex-direction: row;
  height: 50px;
  margin-top: 10px;
  font-size: 24px;
  font-weight: bold;
  justify-content: space-around
}
.card-like {
  margin-left: 40%;
}