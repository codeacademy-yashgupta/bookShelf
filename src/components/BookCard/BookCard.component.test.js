import React from 'react';
import renderer from 'react-test-renderer';
import BookCard from './BookCard.component';

describe('BookCard', () => {
  it('should render book card with respective children', () => {
    const tree = renderer.create(<BookCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
