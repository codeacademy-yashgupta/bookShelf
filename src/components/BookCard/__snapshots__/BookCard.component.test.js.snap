// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard should render book card with respective children 1`] = `
<div
  className="card-body"
>
  <div
    className="card-image"
  >
    <img
      alt=""
      src="./images/cover_image.jpeg"
    />
  </div>
  <div
    className="card-text-title"
  >
    Harry Potter and the Sorcerers Stone (Harry Potter, #1)
  </div>
  <div
    className="card-footer"
  >
    <div
      className="card-text-rating"
    >
      4.5
    </div>
    <div
      className="card-like"
    >
      <div
        className="card-reaction"
      >
        <img
          alt="Press for like"
          className="icon-heart"
          src="./icons/heart-black.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
